tests/integration/test_gil.py

```diff
@@ -14,6 +17,15 @@
 TEST_SINGLE_THREAD_FILE = Path(__file__).parent / "single_thread_program.py"
 
 
+@pytest.fixture(autouse=True)
+def enable_gil_if_free_threading(python, monkeypatch):
+    _, python_executable = python
+    proc = subprocess.run([python_executable, "-Xgil=1", "-cpass"], capture_output=True)
```

Comment:
This will always succeed as an invalid `-X` option doesn't raise or warn. I am missing something here?

Reply:
An unrecognized `-X` option doesn't raise or warn, but `python3.13` recognizes `-Xgil` and fails when the build is not free-threading:
```
$ python3.13 -Xgil=1
Fatal Python error: config_read_gil: PYTHON_GIL / -X gil are not supported by this build
Python runtime state: preinitialized
```

Reply:
I've opened an upstream bug report for this annoying behavior, https://github.com/python/cpython/issues/123275